include/picongpu/fields/currentDeposition/Esirkepov/bitPacking.hpp

```diff
@@ -0,0 +1,90 @@
+/* Copyright 2013-2022 Axel Huebl, Heiko Burau, Rene Widera, Sergei Bastrakov
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "picongpu/fields/currentDeposition/Esirkepov/Line.hpp"
+
+#include <cstdint>
+
+
+namespace picongpu
+{
+    namespace currentSolver
+    {
+        /** Helper base class for Esirkepov to manage bit packing operations to encode the status of the start and end
+         * particle to optimize the algorithm.
+         * Bit-packing for the status is required to keep the register footprint low.
+         */
+        namespace bitpacking
+        {
+            enum class Status
```

Comment:
I didn't think of that before, but perhaps instead of `bitpacking::Status` it could simply be `enum class ParticleStatus` without a namespace. So then the usage is `currentSolver::ParticleStatus` which makes sense.

"bitpacking" is imo too much into details of implementation rather then what it represents

Reply:
I want to keep the namespace separate from other current deposition code parts.